lua/plenary/collections/py_list.lua

```diff
@@ -0,0 +1,157 @@
+-- vim:sw=2
+local List = {}
+
+setmetatable(List, List)
+
+function List:__index(key)
+  if self ~= List then
+    local field = List[key]
+    if field then return field end
+  end
+end
+
+function List:__call(tbl)
```

Comment:
can we add an `List.new` so people can use a constructor in higher order functions. `__call` will just be sugar